### PR TITLE
I've added some tree support (term recursion is already taken by arrays)

### PR DESCRIPTION
**Pull request by danaki**

### CODE REVIEW

**Comment by igorw** [comment]
Why did you change the name? That's probably a bad idea.

**Comment by danaki** [comment]
I've changed it to try on my main project, otherwise I didn't get to make it work with composer. Feel free to incorporate the changes and then I'll drop my fork.

**Comment by igorw** [comment]
@danaki FYI, you should never have to rename the package to make it work with composer. Just add a VCS repo and it should be fine.

**Comment by danaki** [comment]
@igorw I've tried everything until it worked

**Comment by igorw** [comment]
@danaki feel free to ask for help in #composer on freenode. it's likely that there was a different problem. anyway, just saying. ;-)

File: src/Igorw/Silex/ConfigServiceProvider.php [hunk 1]
```diff
@@ -44,7 +44,11 @@ public function register(Application $app)
 
         foreach ($config as $name => $value)
             if ('%' === substr($name, 0, 1))
-                $this->replacements[$name] = (string) $value;
+                $this->replacements[$name] = preg_replace_callback(
+                    '/(%.+%)/U', function($matches) use ($config) {
```
Comment (igorw) [inline src/Igorw/Silex/ConfigServiceProvider.php:48]: This needs to be a bit more robust. What about escaping percentage signs? Which characters are allowed in replacement keys?

**Comment by igorw** [comment]
Closing, since this has not received any attention. If anyone wants to continue working on this, please address the issues mentioned here and open a new PR, thanks.

**Pull request closed without merging**